src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -88,33 +115,75 @@ private void ReflectionReadMembers(object obj, XmlReaderDelegator xmlReader, Xml
+                {
+                    ReflectionReadMember(obj, index, xmlReader, context, members);
+                    memberIndex = index;
+                    requiredIndex = index + 1;
+                }
             }
         }
 
-        private int ReflectionReadMember(object obj, int memberIndex, XmlReaderDelegator xmlReader, XmlObjectSerializerReadContext context, XmlDictionaryString[] memberNames, XmlDictionaryString[] memberNamespaces)
+        // Put all members of the contract (including types from base contract) into 'members'.
+        private int ReflectionGetMembers(ClassDataContract classContract, DataMember[] members)
         {
-            int memberCount = 0;
+            int memberCount = (classContract.BaseContract == null) ? 0 : ReflectionGetMembers(classContract.BaseContract, members);
+            int childElementIndex = memberCount;
+            for (int i = 0; i < classContract.Members.Count; i++, memberCount++)
+            {
+                members[childElementIndex + i] = classContract.Members[i];
+            }
+
+            return memberCount;
+        }
+
+        private void ReflectionReadMember(object obj, int memberIndex, XmlReaderDelegator xmlReader, XmlObjectSerializerReadContext context, DataMember[] members)
+        {
+            DataMember dataMember = members[memberIndex];
 
-            Type classType = _classContract.UnadaptedClassType;
-            int i = memberIndex + memberCount;
-            DataMember dataMember = _classContract.Members[i];
             Type memberType = dataMember.MemberType;
+            if (dataMember.IsGetOnlyCollection)
+            {
+                var memberValue = ReflectionGetMemberValue(obj, dataMember.MemberInfo);
+                context.StoreCollectionMemberInfo(memberValue);
+                ReflectionReadValue(memberType, dataMember.Name, _classContract.StableName.Namespace);
+            }
+            else
+            {
+                var value = ReflectionReadValue(memberType, dataMember.Name, _classContract.StableName.Namespace);
+                MemberInfo memberInfo = dataMember.MemberInfo;
+                if (memberInfo != null)
+                {
+                    ReflectionSetMemberValue(obj, value, memberInfo);
+                }
+                else
+                {
+                    throw new NotImplementedException("PropertyInfo incorrect");
+                }
+            }
+        }
 
-            var value = ReflectionReadValue(memberType, dataMember.Name, _classContract.StableName.Namespace);
-            MemberInfo memberInfo = dataMember.MemberInfo;
-            if (memberInfo != null)
+        private object ReflectionGetMemberValue(object obj, MemberInfo memberInfo)
+        {
+            object memberValue = null;
+            if (memberInfo is PropertyInfo)
+            {
+                PropertyInfo propInfo = (PropertyInfo)memberInfo;
+                memberValue = propInfo.GetValue(obj);
```

Comment:
It's guaranteed that the PropertyInfo we have here has Get method. In NetCore, the Get could be either public or private; In NetNative, we wouldn't get here if the Get was private.

I just added a test case for this.